grr/server/grr_response_server/flows/general/osquery.py

```diff
@@ -19,17 +32,41 @@ class OsqueryFlow(flow_base.FlowBase):
   behaviours = flow_base.BEHAVIOUR_BASIC
 
   args_type = rdf_osquery.OsqueryArgs
+  progress_type = rdf_osquery.OsqueryProgress
+
+  def _UpdateProgress(
+    self,
+    responses: flow_responses.Responses[rdf_osquery.OsqueryResult]
+  ) -> None:
+    # XXX: Less than TRUNCATED_ROWS_COUNT will be used if the original table is
+    # split into chunks smaller than that. It is expected that the limit will
+    # always be smaller than the chunk size, so this shouldn't happen.
+    first_chunk = responses.responses[0]
+    table = first_chunk.table
+
+    self.state.progress.partial_table = _TruncateRows(table)
+    self.state.progress.total_rows_count = len(table.rows)
 
   def Start(self):
     super(OsqueryFlow, self).Start()
+    self.state.progress = rdf_osquery.OsqueryProgress()
+
     self.CallClient(
         server_stubs.Osquery,
         request=self.args,
         next_state=compatibility.GetName(self.Process))
 
-  def Process(self, responses):
+  def Process(
+      self,
+      responses: flow_responses.Responses[rdf_osquery.OsqueryResult]
```

Comment:
Missing trailing comma (it is needed to force auto-formatter not to fold the result type to the same line).